libr/anal/anal.c

```diff
@@ -550,6 +556,7 @@ R_API bool r_anal_noreturn_add(RAnal *anal, const char *name, ut64 addr) {
 			return false;
 		}
 		tmp_name = fcn ? fcn->name: fi->name;
+		anal->is_dirty = true;
```

Comment:
i would move this one line below, inside the is_noreturn.. but maybe only set if noreturn was false before